### ISSUE TITLE
[Android] "refresh-completed" event triggers to early

### ISSUE THREAD

**Issue by svzi** [j3k0/cordova-plugin-purchase#1026]
First of all thanks a lot @j3k0 for this awesome plugin! I really appreciate the work you invest into it, for the past several years! 👍 I'm already using it for quite some time, but today I've encountered an issue.

### system info

```
Cordova:

   Cordova CLI       : 9.0.0 (cordova-lib@9.0.1)
   Cordova Platforms : android 8.1.0, ios 5.1.1
   Cordova Plugins   : cordova-plugin-ionic-keyboard 2.2.0, cordova-plugin-ionic-webview 4.1.3, (and 29 other plugins)

Utility:

   cordova-res (update available: 0.13.1) : 0.8.1
   native-run (update available: 1.0.0)   : 0.2.9

System:

   ios-deploy : 1.9.4
   ios-sim    : 8.0.2
   NodeJS     : v10.17.0 (/Users/svzi/.n/bin/node)
   npm        : 6.14.4
   OS         : macOS Catalina
   Xcode      : Xcode 11.3.1 Build version 11C504

Plugin version: 10.0.1
```

### Expected behavior
I would expect the `refresh-completed` event to trigger, when the validation of the restored purchases has been completed. 

### Observed behavior
The `refresh-completed` event get's triggered, before the restored purchase has been validated by the validator. I wanted to use this event, to hide my progress indicator during the restore process.

Any feedback would be really much appreciated.

**Comment by j3k0** [comment]
First, `refresh-completed` is an internal event. I don't think it's a good idea to rely upon it. Or maybe it is and I should add it to the documentation ;-)

On iOS, it is triggered by this low-level event from StoreKit: `paymentQueueRestoreCompletedTransactionsFinished:`

https://developer.apple.com/documentation/storekit/skpaymenttransactionobserver/1506101-paymentqueuerestorecompletedtran?language=objc

On Android, it is triggered when `getPurchases()` returns the list of purchased products.

At higher level, on both cases, I think this means there are no more products to send to the `approved` state.

Anyway. you could monitor the state of the products and see if there are still transactions under processing.

```js
function monitorPurchases() {
  if (store.products.filter(p => p.state === 'approved').length === 0) {
    // done processing
    store.off(monitorPurchases);
    hideProgressIndicator();
  }
}
store.when().updated(monitorPurchases);
showProgressIndicator();
store.refresh();
```

Untested, but should work on all platforms.

(bonus idea). You can also use the current number of products in the `approved` state as a way to estimate "progress". When it increases, you're getting closer to 50%. When it decreases, you're getting closer to 100%.

**Comment by svzi** [comment]
@j3k0 Thanks for the ultra fast reply! Really appreciate your support. 👍 

As I said earlier, I'm using your plugin quite some time, but today I've stumbled upon the `refresh-complete` event, because you mentioned it in some issues that I found before I created this one. Before that I didn't even knew that even exists. 

But anyway, I've just tried your suggested solution and it works perfectly well. Thanks a lot again, and keep up the great work!
